### ISSUE TITLE
mcrypt is removed in PHP 7.2

### ISSUE THREAD

**Issue by craigcarnell** [Inchoo/Inchoo_PHP7#126]
https://wiki.php.net/rfc/mcrypt-viking-funeral

https://wiki.php.net/rfc/libsodium

I haven't tested anything in 7.2 yet..

https://github.com/Inchoo/Inchoo_PHP7/issues/98

**Comment by mklooss** [comment]
you will also some problems with the customer area in the backend:

`PHP Fatal error:  Declaration of Mage_Customer_Model_Resource_Address_Attribute_Source_Country::getAllOptions() must be compatible with Mage_Eav_Model_Entity_Attribute_Source_Table::getAllOptions($withEmpty = true, $defaultValues = false) in magento/app/code/core/Mage/Customer/Model/Resource/Address/Attribute/Source/Country.php on line 0`
Testet in Magento 1.9.3.7

i've fixed this with "rewrites"

https://github.com/mklooss/Loewenstark_Php72

**Comment by postadelmaga** [comment]
Hi guys,
Since I was planing a switch to PHP 7.X I am also interested into this fix.
.. And at this point makes sense to look at PHP 7.2 issues before doing such a switch from PHP 5.6 ...

Personally I don't know much about Mcrypt, I suppose it used mostly to crypt password stored in DB but there is probably something more ( otherwise why just switch to a different cryptography library all the password ? I mean an extension, to be installed before switching to PHP 7.2 that migrate the passwords from mcrypt to the new lib ? ) 

At the moment I can only give the following contribution:  
- Some alternatives to MCrypt are listed in this [Stack Overflow Question](https://stackoverflow.com/a/41272680) 

- There isan open issue on for [Magento 2](https://github.com/magento/magento2/issues/12273#issuecomment-354912180), I guess some of the code can be donwported to M1

I have experience on M1 but not much with this part of the PHP libraries anyway I am glad to help anyone that has a better understanding of the issue, just let me know how :) ( for sure I am glad to test any patch/extension to fix this on my codebase )

**Comment by durzel** [comment]
Out of interest - what is the goal by upgrading to 7.2?  There is a tangible speed benefit from going from 5.6.x to 7.0, and this extension works just fine with 7.0.

**Comment by hostep** [comment]
@durzel: PHP 7.0 is [EOL](http://php.net/supported-versions.php) at the end of 2018, which is not so far away anymore.

**Comment by sr972** [comment]
So still PHP 7.1 left to use with MCrypt.
BTW, MCrypt is still available in 7.2 via PECL. If this is the best choice? Don't know.

**Comment by ivanweiler** [comment]
Related to #98 , I wrote small status update there

**Comment by craigcarnell** [comment]
@ivanweiler looking forward to the release, when is it expected?

**Comment by mitcht** [comment]
Update... As of today, the Ubuntu 18 Daily build has dropped all built in support for packages for php7.1.

**Comment by mitcht** [comment]
The development branch is seeing some activity as of 8 days ago. https://github.com/Inchoo/Inchoo_PHP7/tree/develop

**Comment by craigcarnell** [comment]
Any updates on this? I've now install php7.2 locally and upgraded my OS to Ubuntu 18.04 due to some hardware support issues.

**Comment by mitcht** [comment]
I've found that installing mcrypt via pecl is still a viable solution at this time.

**Comment by postadelmaga** [comment]
Magento 2 community has created a project to fix all problems related PHP 7.2:
- https://github.com/magento-engcom/php-7.2-support

It can be useful, I am pretty sure we can backport some of the solution adopted there ...  in case there are enough folks here that want to keep Magento 1 alive ...

**Comment by ivanweiler** [comment]
Last time I checked (a while ago) M2 was still using mcrypt, with suppressed warnings, so they're not yet 7.2 ready:
https://github.com/magento-engcom/php-7.2-support/blob/2.3-develop/lib/internal/Magento/Framework/Encryption/Crypt.php#L59

But we don't mind if they backport some of the solutions from us ;)

Our M1 solution can be seen in our current develop branch, feel free to test, more info here
https://github.com/Inchoo/Inchoo_PHP7/issues/98#issuecomment-363050865

We just can't find some free time to do final release, so if anyone can test a bit it would be really helpful !! We are concentrating on php 7.1, but everything implemented will solve most of 7.2 issues.

**Comment by piotrekkaminski** [comment]
Anyone having luck using this with 7.2? What is your experience? What types of warnings get generated?

**Comment by craigcarnell** [comment]
Any updates on this? Is there going to be an official Magento 1.x release? as per this:

```
We're currently working with Magento directly on releasing the official PHP 7.2 patch. This will include some changes and lot of additional testing across versions, so we ask for a bit more of your patience. Hopefully, version 3.0 of this extension won't be needed anymore :)
```

**Comment by mrlerch** [comment]
That is absolutely awesome! Great work and keep it up.

Martin

**Comment by mitcht** [comment]
Agreed. This is good news. It will simplify my automated deployment script to not require pecl and the required building.

**Comment by durzel** [comment]
I'm confused.. @craigcarnell asked for an update on developments, he wasn't saying there had been progress made?

**Comment by mitcht** [comment]
My thoughts taken from the comment:

> We're currently working with Magento directly on releasing the official PHP 7.2 patch.
I take it inchoo is working with magento for a fix for mcrypt to go into core.

>  This will include some changes and lot of additional testing across versions, so we ask for a bit more of your patience. Hopefully, version 3.0 of this extension won't be needed anymore :)
Across versions = 1 & 2. The Inchoo_PHP7 won't be needed any longer.

**Comment by durzel** [comment]
The "We're currently working with Magento" comment is from [here](https://github.com/Inchoo/Inchoo_PHP7/wiki/PHP-7.1-and-7.2), posted back in June.

There may have been developments in that time, which is what @craigcarnell was asking for an update on I think.  To the best of my knowledge the develop branch (3.0.0) has fixes for PHP 7.1 and 7.2, but it's "untested" per @ivanweiler's earlier comment.

**Comment by mitcht** [comment]
It really is, but I may be a candidate to do some testing soon. we are do to revisit our ecommerce site in the next few months. I may introduce that into our test cycle.  I will need to review the code with our auditor to ensure there aren't any glaring security flaws but I don't see any myself.
